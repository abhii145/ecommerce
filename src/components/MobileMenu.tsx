import React from "react"
import { Link } from "react-router"

interface MobileMenuProps {
  setMenuOpen: React.Dispatch<React.SetStateAction<boolean>>
}

const MobileMenu: React.FC<MobileMenuProps> = ({ setMenuOpen }) => {
  const handleLinkClick = () => setMenuOpen(false)

  return (
    <nav
      className="md:hidden bg-white shadow-md p-4 space-y-4"
      aria-label="Mobile Menu"
    >
      <Link
        to="/products"
        onClick={handleLinkClick}
        className="block hover:text-blue-500 transition-colors"
      >
        Products
      </Link>
      <Link
        to="/favourites"
        onClick={handleLinkClick}
        className="block hover:text-blue-500 transition-colors"
      >
        Favourites
      </Link>
      <Link
        to="/orders"
        onClick={handleLinkClick}
        className="block hover:text-blue-500 transition-colors"
      >
        Orders
      </Link>
      <Link
        to="/cart"
        onClick={handleLinkClick}
        className="block hover:text-blue-500 transition-colors"
      >
        Cart
      </Link>
    </nav>
  )
}

export default React.memo(MobileMenu)
