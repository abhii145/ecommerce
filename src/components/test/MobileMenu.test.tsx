import { render, screen, fireEvent } from "@testing-library/react"
import { BrowserRouter as Router } from "react-router"
import MobileMenu from "../MobileMenu"

describe("MobileMenu component", () => {
  it("matches snapshot", () => {
    const setMenuOpen = jest.fn()
    const { asFragment } = render(
      <Router>
        <MobileMenu setMenuOpen={setMenuOpen} />
      </Router>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders MobileMenu component", () => {
    const setMenuOpen = jest.fn()

    render(
      <Router>
        <MobileMenu setMenuOpen={setMenuOpen} />
      </Router>
    )

    expect(screen.getByText(/Products/i)).toBeInTheDocument()
    expect(screen.getByText(/Favourites/i)).toBeInTheDocument()
    expect(screen.getByText(/Orders/i)).toBeInTheDocument()
    expect(screen.getByText(/Cart/i)).toBeInTheDocument()
  })

  it("closes menu on link click", () => {
    const setMenuOpen = jest.fn()

    render(
      <Router>
        <MobileMenu setMenuOpen={setMenuOpen} />
      </Router>
    )

    fireEvent.click(screen.getByText(/Products/i))
    expect(setMenuOpen).toHaveBeenCalledWith(false)

    fireEvent.click(screen.getByText(/Favourites/i))
    expect(setMenuOpen).toHaveBeenCalledWith(false)

    fireEvent.click(screen.getByText(/Orders/i))
    expect(setMenuOpen).toHaveBeenCalledWith(false)

    fireEvent.click(screen.getByText(/Cart/i))
    expect(setMenuOpen).toHaveBeenCalledWith(false)
  })
})
