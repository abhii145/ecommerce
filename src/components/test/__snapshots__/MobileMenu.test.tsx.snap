// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenu component matches snapshot 1`] = `
<DocumentFragment>
  <nav
    aria-label="Mobile Menu"
    class="md:hidden bg-white shadow-md p-4 space-y-4"
  >
    <a
      class="block hover:text-blue-500 transition-colors"
      data-discover="true"
      href="/products"
    >
      Products
    </a>
    <a
      class="block hover:text-blue-500 transition-colors"
      data-discover="true"
      href="/favourites"
    >
      Favourites
    </a>
    <a
      class="block hover:text-blue-500 transition-colors"
      data-discover="true"
      href="/orders"
    >
      Orders
    </a>
    <a
      class="block hover:text-blue-500 transition-colors"
      data-discover="true"
      href="/cart"
    >
      Cart
    </a>
  </nav>
</DocumentFragment>
`;
